src/main/java/com/gmail/inverseconduit/bot/InteractionBot.java

```diff
@@ -26,13 +26,12 @@ public InteractionBot(ChatInterface chatInterface) {
 
     @Override
     public void start() {
-        executor.scheduleAtFixedRate(this::processInteractions, 200, 700, TimeUnit.MILLISECONDS);
+        // asynchronously enqueue the processing by blocking supplier
+        processingThread.submit(this::processInteractions);
```

Comment:
Same here; a `ScheduledExecutorService` conveys the intent better. Abstraction has limits ;)